app/scripts/services/dimBungieService.factory.js

```diff
@@ -60,20 +60,49 @@
         return $q.reject(new Error($translate.instant('BungieService.Down')));
       }
       if (response.status < 200 || response.status >= 400) {
-        return $q.reject(new Error($translate.instant('BungieService.NetworkError', { status: response.status, statusText: response.statusText })));
+        return $q.reject(new Error($translate.instant('BungieService.NetworkError', {
+          status: response.status,
+          statusText: response.statusText
+        })));
       }
 
       var errorCode = response.data.ErrorCode;
+
+      switch (errorCode) {
+      case 1: {
+        return response;
+      }
+      case 1627: {
+        return $q.reject("Vendor data is unavailable.");
+      }
+      case 2108: {
+        $rootScope.$broadcast('dim-no-token-found');
+        return $q.reject("DIM does not have permission to perform this action.");
+      }
+      case 5:
+      case 36:
+      case 99:
+      case 1618:
+      case 2101:
+      case 2102:
+      case 2107:
```

Comment:
That is a stepping stone towards using an object literal as the control flow.

https://toddmotto.com/deprecating-the-switch-statement-for-object-literals/